api/lib/opentelemetry/instrumentation/base.rb

```diff
@@ -129,14 +129,27 @@ def compatible(&blk)
           @compatible_blk = blk
         end
 
+        # The option method is used to define default configuration options
+        # for the instrumentation library. It requires a name, default value,
+        # and a validation callable to be provided.
+        # @param [String] name The name of the configuration option
+        # @param default The default value to be used, or to used if validation fails
+        # @param [Callable] validate A callable that validates the option value and returns a boolean
```

Comment:
We may want to relax this at some point and make the `validate` Callable optional, but I think it's fine to keep this restrictive for now.

Reply:
This comment should be expanded to include the built-in validators.